tracer/src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -119,7 +119,7 @@ public static bool TryConvertToNormalizedTagName(this string value, bool normali
                 {
                     case (>= 'a' and <= 'z') or (>= '0' and <= '9') or '_' or ':' or '/' or '-':
                         continue;
-                    case '.' when !normalizePeriods:
+                    case '.' or ' ' when !normalizePeriodsAndSpaces:
```

Comment:
I think we should _always_ normalize periods (i.e. the "old" featureflag=true) behaviour, as noone ever disabled that. Which means the logic would become something like this (I renamed the variable too)
```suggestion
                    case '.':
                        continue;
                     case ' ' when !normalizeAndSpaces:
```

Reply:
Agreed, thanks. I'm fixing your code so the `.` falls into the `default` case, though.